### PR TITLE
[vcr-2.0] Impl TOKEN

### PR DESCRIPTION
**Pull request by snalli**
Implements token persist and retrieve

### CODE REVIEW

File: ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureCloudDestinationSync.java [hunk 1]
```diff
@@ -625,16 +625,76 @@ public int compactPartition(String partitionPath) throws CloudStorageException {
...
+    String azureTokenFileName = getAzureTokenFileName(tokenLayout, tokenFileName);
+    // There is no parallelism, but we still need to create and pass this object to SDK.
+    BlobParallelUploadOptions blobParallelUploadOptions =
+        new BlobParallelUploadOptions(inputStream);
+    // Without content-type, get-token floods log with warnings
+    blobParallelUploadOptions.setHeaders(new BlobHttpHeaders().setContentType("application/octet-stream"));
+    try {
+      BlobContainerClient blobContainerClient = createOrGetBlobStore(TOKEN_CONTAINER);
+      ////////////////////////////////// Upload token to Azure blob storage ////////////////////////////////////////
+          blobContainerClient.getBlobClient(azureTokenFileName)
```
Comment (JingQianCloud) [inline ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureCloudDestinationSync.java:668]: nit: format

File: ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureCloudDestinationSync.java [hunk 2]
```diff
@@ -625,16 +625,76 @@ public int compactPartition(String partitionPath) throws CloudStorageException {
...
       throws CloudStorageException {
-    return false;
+    // Prepare to download token from Azure blob storage
+    AzureBlobLayoutStrategy.BlobLayout
+        tokenLayout = azureBlobLayoutStrategy.getTokenBlobLayout(partitionPath, tokenFileName);
+    String azureTokenFileName = getAzureTokenFileName(tokenLayout, tokenFileName);
+    try {
+      BlobContainerClient blobContainerClient = createOrGetBlobStore(TOKEN_CONTAINER);
+      ////////////////////////////////// Download token from Azure blob storage ////////////////////////////////////////
+      blobContainerClient.getBlobClient(azureTokenFileName)
```
Comment (JingQianCloud) [inline ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureCloudDestinationSync.java:689]: nit: format

**Review by JingQianCloud — APPROVED** [review]
overall looks good to me. please do a reformat before the checkin.

**Review by Arun-LinkedIn — APPROVED** [review]

File: ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureCloudDestinationSync.java (hunk 1 above)
Comment (snalli) [reply ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureCloudDestinationSync.java]: done

File: ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureCloudDestinationSync.java (hunk 2 above)
Comment (snalli) [reply ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureCloudDestinationSync.java:689]: done

**Comment by snalli** [comment]
Merging as all comments addressed and no more pending issues

**Pull request merged**